### PR TITLE
Support context.Context

### PR DESCRIPTION
**Pull request by haya14busa**
Also, update go-github

Fixes: #80

### CODE REVIEW

File: comment_iowriter.go [hunk 1]
```diff
@@ -18,7 +19,7 @@ func NewRawCommentWriter(w io.Writer) *RawCommentWriter {
 	return &RawCommentWriter{w: w}
 }
 
-func (s *RawCommentWriter) Post(c *Comment) error {
+func (s *RawCommentWriter) Post(_ context.Context, c *Comment) error {
```
Comment (reviewdog) [inline comment_iowriter.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method RawCommentWriter.Post should have comment or be unexported

File: comment_iowriter.go [hunk 2]
```diff
@@ -41,7 +42,7 @@ func NewUnifiedCommentWriter(w io.Writer) *UnifiedCommentWriter {
 	return &UnifiedCommentWriter{w: w}
 }
 
-func (mc *UnifiedCommentWriter) Post(c *Comment) error {
+func (mc *UnifiedCommentWriter) Post(_ context.Context, c *Comment) error {
```
Comment (reviewdog) [inline comment_iowriter.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method UnifiedCommentWriter.Post should have comment or be unexported

File: diff.go [hunk 3]
```diff
@@ -16,7 +17,7 @@ func NewDiffString(diff string, strip int) DiffService {
 	return &DiffString{b: []byte(diff), strip: strip}
 }
 
-func (d *DiffString) Diff() ([]byte, error) {
+func (d *DiffString) Diff(_ context.Context) ([]byte, error) {
```
Comment (reviewdog) [inline diff.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method DiffString.Diff should have comment or be unexported

**Comment by JensRantil** [comment]
Just stumbled across this pull request. I don't see that the contexts are being used for anything in this patch. Why add it?

**Comment by haya14busa** [comment]
it's used for GitHub request

> `g.cli.PullRequests.CreateReview(ctx, g.owner, g.repo, g.pr, review)`

**Comment by JensRantil** [comment]
Ah, missed that one! But the context is never modified nor cancelled. Is it worth having it sprinkled all over if it has no use? A different, less invasive, approach would be to submit `context.Background()` to all Github client calls.

**Pull request merged**